### PR TITLE
feat(dashboard): Issue widgets count and timestamp style and format

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Added tooltip to issue count fields and formatted timestamp fields
This PR also involves some refactoring out issue fields logic from `fieldRenderers` and moving them to their own file at `issueFieldRenderers`
~~TODO: Should refactor the issues stuff out of `fieldrenderers`~~ Done
TODO: Need to make count tooltip numbers clickable
image

### CODE REVIEW

**Comment by edwardgou-sentry** [comment]
Drafting this for now since I want refactor Issues stuff out of `fieldRenderers.tsx` first before merging this in.

**Comment by edwardgou-sentry** [comment]
Added the issue fields refactor into this pr and reopened

**Comment by narsaynorath** [comment]
The tooltip on the UserCount renders as expected (with the total count of users for the last 24 hrs and then the lifetime total)
Screen Shot 2022-01-18 at 10 27 51 AM

But I noticed the tooltip also appears on the LifetimeUserCount with the count numbers, not the user numbers
Screen Shot 2022-01-18 at 10 26 34 AM

Do we want to fix that in this PR?

File: static/app/utils/dashboards/issueFieldRenderers.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  </div>
+))`
+  text-align: right;
+  font-weight: bold;
+  font-variant-numeric: tabular-nums;
+  padding-left: ${space(2)};
+  color: ${p => p.theme.subText};
+`;
+
+const Divider = styled('li')`
```
Comment (narsaynorath) [inline static/app/utils/dashboards/issueFieldRenderers.tsx:218]: This can probably be a div, I think even for a11y it wouldn't provide any useful info

**Comment by edwardgou-sentry** [comment]
> The tooltip on the UserCount renders as expected (with the total count of users for the last 24 hrs and then the lifetime total) Screen Shot 2022-01-18 at 10 27 51 AM
> 
> But I noticed the tooltip also appears on the LifetimeUserCount with the count numbers, not the user numbers Screen Shot 2022-01-18 at 10 26 34 AM
> 
> Do we want to fix that in this PR?

Oh nice catch, this is due to a capitalization issue which I fixed using regex as you suggested!

File: static/app/views/dashboardsV2/widgetCard/issueWidgetQueries.tsx [hunk 2]
```diff
@@ -17,7 +19,7 @@ import {Widget, WidgetQuery} from '../types';
 const MAX_ITEMS = 5;
 const DEFAULT_SORT = IssueSortOptions.DATE;
 const DEFAULT_DISPLAY = IssueDisplayOptions.EVENTS;
-const DEFAULT_COLLAPSE = ['filtered'];
+const DEFAULT_COLLAPSE = [];
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/views/dashboardsV2/widgetCard/issueWidgetQueries.tsx:22]: Endpoint will still work without this being passed, but you're probably right, I don't think I'll be adding any new collapses in the future so I just removed it entirely

File: static/app/utils/dashboards/issueFieldRenderers.tsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+const SecondaryCount = styled(Count)`
+  :before {
+    content: '/';
+    padding-left: ${space(0.25)};
+    padding-right: 2px;
+  }
+`;
+
+const ItemCount = styled(({value, ...p}) => (
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/utils/dashboards/issueFieldRenderers.tsx:206]: The `...p` is used for styling via adding a class to the `div` in this component!

File: static/app/components/charts/simpleTableChart.tsx [hunk 4]
```diff
@@ -29,10 +32,18 @@ class SimpleTableChart extends Component<Props> {
     tableMeta: NonNullable<TableData['meta']>,
     columns: ReturnType<typeof decodeColumnOrder>
   ) {
-    const {location, organization} = this.props;
+    const {location, organization, source} = this.props;
 
     return columns.map(column => {
-      const fieldRenderer = getFieldRenderer(column.name, tableMeta);
+      let fieldRenderer;
+      switch (source) {
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/components/charts/simpleTableChart.tsx:39]: Good call! I've updated `simpleTableChart` to allow injecting an optional `getCustomFieldRenderer` via props and removed the `source` prop!

File: static/app/utils/dashboards/issueFieldRenderers.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+
+const contentStyle = css`
+  width: 100%;
+  justify-content: space-between;
+  display: flex;
+  padding: 6px 10px;
+`;
+
+const StyledContent = styled('div')`
+  ${contentStyle};
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/utils/dashboards/issueFieldRenderers.tsx:195]: I think I may be using `contentStyle` for another component in a future pr soon so I'd like to keep this here for now. If I end up not needing to reuse this css, will definitely just move it into `StyledContent` next update!

File: static/app/utils/dashboards/issueFieldRenderers.tsx [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+    renderFunc: ({lastSeen}) => <StyledDateTime date={lastSeen} />,
+  },
+};
+
+const issuesCountRenderer = (
+  data: EventData,
+  field: 'count' | 'userCount' | 'lifetimeCount' | 'lifetimeUserCount'
+) => {
+  const {selectionDateString} = data;
+  const isUserField = field.includes('user');
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/utils/dashboards/issueFieldRenderers.tsx:128]: yup this is correct! updated to use regex with the ignore caps flag

File: static/app/utils/dashboards/issueFieldRenderers.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/utils/dashboards/issueFieldRenderers.tsx:218]: Yup, updated to div!

File: static/app/utils/dashboards/issueFieldRenderers.tsx [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+          ) : (
+            <Count value={primaryCount} />
+          )}
+        </span>
+      </Tooltip>
+    </Container>
+  );
+};
+
+const Item = styled('li')`
```
[the remark this answers is not in the document]
Comment (edwardgou-sentry) [reply static/app/utils/dashboards/issueFieldRenderers.tsx]: Yeah `Item` and `li` is actually not necessary here, it was leftover attempting the tooltip with a different parent component. Removed it!

**Review by narsaynorath — APPROVED** [review]
Tested it locally and it looks good to me!

Looks like there's a pretty big visual diff. I don't think any of these changes affect the Issue detail page so maybe it just needs master to get merged back in?

**Pull request merged**